src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowPackage.cs

```diff
@@ -30,22 +28,24 @@ internal abstract partial class VsInteractiveWindowPackage<TVsInteractiveWindowP
         private IComponentModel _componentModel;
         private TVsInteractiveWindowProvider _interactiveWindowProvider;
 
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
+            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
 
             // Load the Roslyn package so that its FatalError handlers are hooked up.
-            var shell = (IVsShell)this.GetService(typeof(SVsShell));
+            var shell = (IVsShell)await GetServiceAsync(typeof(SVsShell)).ConfigureAwait(true);
             shell.LoadPackage(Guids.RoslynPackageId, out var roslynPackage);
             
             // Explicitly set up FatalError handlers for the InteractiveWindowPackage.
             SetErrorHandlers(typeof(IInteractiveWindow).Assembly);
             SetErrorHandlers(typeof(IVsInteractiveWindow).Assembly);
 
-            _componentModel = (IComponentModel)GetService(typeof(SComponentModel));
+            _componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(true);
             _interactiveWindowProvider = _componentModel.DefaultExportProvider.GetExportedValue<TVsInteractiveWindowProvider>();
```

Comment:
Querying MEF is often best done from a background thread since it often results in loading assemblies.
But if you do this, keep risk down by verifying that the export(s) that are activated by this line of code don't have thread affinity hiding in their importing constructors.